servicetalk-http-router-predicate/src/main/java/io/servicetalk/http/router/predicate/InOrderRouter.java

```diff
@@ -67,7 +68,10 @@
                 StreamingHttpService service = pair.service();
                 final HttpExecutionStrategy strategy = pair.routeStrategy();
                 if (strategy != null) {
-                    service = strategy.offloadService(ctx.executionContext().executor(), service);
+                    Executor executor = null == strategy.executor() ?
```

Comment:
`strategy.executor()` in theory may change in-between checking against null and then saving the reference. cache the reference in a local variable to avoid this.

Reply:
also isn't the plan to remove `executor()` from the strategy? why do we need to change this code to use the `strategy`'s executor?

Reply:
It was changed for consistency. Yes, it will soon be obsolete.